### DISCUSSION TITLE
Jstree, colouring a text node

### DISCUSSION THREAD

**Discussion by Pete6666** [vakata/jstree#2521] [Q&A] [answered]
I want to colour the text of a Jstree node based either on the id or any other field in the source Mysql table.

My jstree gets its data from a mysql table with 3 fields, here is the table structure and data with just one record:

Field names are: id, notabledate, parent_id

Data for this record is: red-68256170069 2021-04-19 #

Here is the code for the tree:
function buildeditnoteabledatestreeview()
    {               
        $('#noteabledatestree').jstree('refresh');                          
        $('#noteabledatestree').jstree(
        {
            
        plugins: ["wholerow",  "json_data", "ui", "themes",  "sort", "checkbox", "types"],
                'core' : 
                {
                    'multiple': false,
                    "themes":
                    {                       
                       "icons":false,                     
                       "responsive": false,
                       "stripes": false,
                    },
                                        
                    "check_callback": true,
                                        
                    'data' : 
                    {
                      "url" : "tl2_get_allnoteabledates_as_tree.php",                   
                      "dataType" : "json"               
                    }
                },
                
                'checkbox':                
                {       
                      'three_state'   : false,                  
                      'tie_selection' : false,
                      'deselect_all'  : true,                 
                },
                
        })
    } 

The tree is built ok, and looks like this:

☐ 2021-04-19

☐ 2021-04-18

What I what to do is use the prefix in the id (ie red, green etc) to cause the text of the node (a date eg 2021-04-18) to be coloured. So for example the date 2021-04-19 would have the text colour red. Any help appreciated.

**Comment by vakata** [comment, marked as answer]
Using CSS will be fastest (and easiest):
```
#noteabledatestree [id^="red"] .jstree-anchor { color:red; }
```

**Reply by Pete6666** [reply]
Thank you for your response.
